tests/e2e/volumeproperty_test.go

```diff
@@ -0,0 +1,176 @@
+	out, err = ssh.InvokeCommand(s.config.DockerHosts[1], s.dockerVolumeRemoveCmd)
+	c.Assert(err, IsNil, Commentf(out))
+	log.Println(" CLEAN-UP Finished: volumeproperty_test.TestVolumeProperties")
+}
+
+var _ = Suite(&VolumePropertyTestSuite{})
+var dockerCliMap = make(map[string][]string)
+var adminCliMap = make(map[string][]string)
+var expectedValues = make(map[string][]string)
+
+/*
+Test steps:
+1. SSH to a vm and create volumes of size 100 mb and specified disk format ["thin", "zeroedthick", "eagerzeroedthick"]
+2. Do docker inspect on the volume get size, disk format and attached-to-vm field.
+   Expected value: {"100MB", "thin/zeroedthick/eagerzeroedthick", no-value} respectively.
+3. SSH to the esx and get capacity, disk format and attached-to-vm field for the volume using Admin cli
+   Expected value: {"100MB", "thin/zeroedthick/eagerzeroedthick", "detached"} respectively.
+4. Verifies values obtained from admin cli and docker cli.
+5. SSH to the vm and run a container and mount the volume created.
+6. SSH to the vm and esx and verify the attached-to-vm field for volume - both docker cli and admin cli values should be same.
+7. Again verify capacity and disk format using docker cli and admin cli to make sure things are fine after running the container.
+
+NOTE: Do steps 5 ,6 and 7 only for volume of 'thin' disk format
+*/
+
+func (s *VolumePropertyTestSuite) TestVolumeProperties(c *C) {
+	log.Printf("START: volumeproperty_test.TestVolumeProperties")
+
+	// create volumes of all three disk formats
+	s.createVolumes(c)
+
+	// get properties for all three volumes from ESX and docker host
+	s.getVolumeProperties(c)
+
+	// Verify if docker and ESX properties of all three volumes are same and as expected.
+	s.verifyProperties(c)
+
+	// attach only thin volume
+	out, err := dockercli.AttachVolume(s.config.DockerHosts[1], s.volumeNames[0], s.containerName)
+	c.Assert(err, IsNil, Commentf("Failed to attach the volume [%s]", out))
+	s.volumeAttached = true
+
+	// Get volumes properties for the thin volume from docker and ESX.
+	s.getVolumePropertiesAdminCli(c, s.volumeNames[0])
+	s.getVolumePropertiesDockerCli(c, s.volumeNames[0])
+
+	// Verify if docker and ESX properties of volumes are same and as expected.
+	s.verifyProperties(c)
+
+	log.Printf("END: volumeproperty_test.TestVolumeProperties")
+}
+
+// createVolumes - creates volumes of each format type
+func (s *VolumePropertyTestSuite) createVolumes(c *C) {
+	for _, formatType := range s.formatTypes {
+		vname := inputparams.GetVolumeNameWithTimeStamp(formatType)
+		out, err := dockercli.CreateVolumeWithOptions(s.config.DockerHosts[1], vname, diskFormatOption+formatType)
+		s.volumeNames = append(s.volumeNames, vname)
+		c.Assert(err, IsNil, Commentf(out))
+	}
+}
+
+// getVolumeProperties - get properties of three volumes from ESX and docker host
+func (s *VolumePropertyTestSuite) getVolumeProperties(c *C) {
```

Comment:
as this method is not returning anything .. we should avoid using `getXXX`; it spoils objective of `get` (leads to confusion)